python/src/trezorlib/stellar.py

```diff
@@ -33,309 +55,215 @@
+    elif isinstance(memo, ReturnHashMemo):
+        tx.memo_type = MEMO_TYPE_RETURN
+        tx.memo_hash = memo.memo_return
+    else:
+        tx.memo_type = MEMO_TYPE_NONE
+
+    tx.num_operations = len(parsed_tx.operations)
     operations = []
-    for _ in range(tx.num_operations):
-        operations.append(_parse_operation_bytes(unpacker))
+    for parsed_operation in parsed_tx.operations:
+        operations.append(_read_operation(parsed_operation))
 
     return tx, operations
 
 
-def _parse_operation_bytes(unpacker):
-    """Returns a protobuf message representing the next operation as read from
-    the byte stream in unpacker
-    """
-
-    # Check for and parse optional source account field
-    source_account = None
-    if unpacker.unpack_bool():
-        source_account = unpacker.unpack_fopaque(32)
-
-    # Operation type (See OP_ constants)
-    type = unpacker.unpack_uint()
-
-    if type == OP_CREATE_ACCOUNT:
+def _read_operation(parsed_operation: Operation):
+    # TODO: Let's add muxed account support later.
+    source_account = parsed_operation.source
+    if source_account:
+        source_account = source_account.account_id
+    if isinstance(parsed_operation, CreateAccount):
         return messages.StellarCreateAccountOp(
             source_account=source_account,
-            new_account=_xdr_read_address(unpacker),
-            starting_balance=unpacker.unpack_hyper(),
+            new_account=parsed_operation.destination,
+            starting_balance=_read_amount(parsed_operation.starting_balance),
         )
-
-    if type == OP_PAYMENT:
+    if isinstance(parsed_operation, Payment):
         return messages.StellarPaymentOp(
             source_account=source_account,
-            destination_account=_xdr_read_address(unpacker),
-            asset=_xdr_read_asset(unpacker),
-            amount=unpacker.unpack_hyper(),
+            destination_account=parsed_operation.destination.account_id,
+            asset=_read_asset(parsed_operation.asset),
+            amount=_read_amount(parsed_operation.amount),
         )
-
-    if type == OP_PATH_PAYMENT:
-        op = messages.StellarPathPaymentOp(
+    if isinstance(parsed_operation, PathPaymentStrictReceive):
+        operation = messages.StellarPathPaymentOp(
             source_account=source_account,
-            send_asset=_xdr_read_asset(unpacker),
-            send_max=unpacker.unpack_hyper(),
-            destination_account=_xdr_read_address(unpacker),
-            destination_asset=_xdr_read_asset(unpacker),
-            destination_amount=unpacker.unpack_hyper(),
+            send_asset=_read_asset(parsed_operation.send_asset),
+            send_max=_read_amount(parsed_operation.send_max),
+            destination_account=parsed_operation.destination.account_id,
+            destination_asset=_read_asset(parsed_operation.dest_asset),
+            destination_amount=_read_amount(parsed_operation.dest_amount),
             paths=[],
```

Comment:
would this work? `paths=[_read_asset(asset) for asset in operation.paths]`

Reply:
75ebbce0d081dde71c79022c1822943c892efe32